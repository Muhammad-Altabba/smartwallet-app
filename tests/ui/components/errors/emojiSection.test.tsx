import React from 'react'
import {
  Emoji,
  EmojiButton,
  EmojiSection,
} from '../../../../src/ui/errors/components/emojiSection'
import { shallow } from 'enzyme'
import { View } from 'react-native'

describe('EmojiSection Component', () => {
  const defaultProps = {
    selectedEmoji: Emoji.Empty,
    setEmoji: jest.fn(),
  }

  const defaultButtonProps = {
    emoji: Emoji.Shit,
    isSelected: false,
    onPress: jest.fn(),
    areAnySelected: false,
  }

  it('matches the snapshot on initial render', () => {
    const component = shallow(<EmojiSection {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('renders the emojis when one is selected', () => {
    const props = { ...defaultProps, selectedEmoji: Emoji.Shit }
    const component = shallow(<EmojiSection {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('matches the snapshot to initial EmojiButton render', () => {
    const component = shallow(<EmojiButton {...defaultButtonProps} />)
    expect(component).toMatchSnapshot()
  })

  it('renders selected emoji button', () => {
    const props = {
      ...defaultButtonProps,
      isSelected: true,
      areAnySelected: true,
    }
    const component = shallow(<EmojiButton {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('calls onPress when Emoji is pressed', () => {
    const component = shallow(<EmojiButton {...defaultButtonProps} />)
    component.find(View).simulate('touchEnd')
    expect(defaultButtonProps.onPress).toBeCalledTimes(1)
  })
})
