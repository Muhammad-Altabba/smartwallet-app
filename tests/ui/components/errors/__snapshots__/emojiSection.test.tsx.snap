// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiSection Component matches the snapshot on initial render 1`] = `
<Fragment>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "space-between",
        "marginTop": 27,
        "maxWidth": "100%",
      }
    }
  >
    <EmojiButton
      areAnySelected={true}
      emoji="💩"
      isSelected={false}
      key="0"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="😘"
      isSelected={false}
      key="1"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="🤦‍♀"
      isSelected={false}
      key="2"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="👿"
      isSelected={false}
      key="3"
      onPress={[Function]}
    />
  </View>
</Fragment>
`;

exports[`EmojiSection Component matches the snapshot to initial EmojiButton render 1`] = `
<View
  onTouchEnd={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "borderColor": "rgb(62, 62, 62)",
      "borderRadius": 50,
      "borderWidth": 1,
      "height": 66,
      "justifyContent": "center",
      "width": 66,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 31,
      }
    }
  >
    💩
  </Text>
</View>
`;

exports[`EmojiSection Component renders selected emoji button 1`] = `
<View
  onTouchEnd={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.5)",
      "borderColor": "rgb(197, 41, 105)",
      "borderRadius": 50,
      "borderWidth": 1,
      "height": 66,
      "justifyContent": "center",
      "width": 66,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 31,
      }
    }
  >
    💩
  </Text>
</View>
`;

exports[`EmojiSection Component renders the emojis when one is selected 1`] = `
<Fragment>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": "auto",
        "justifyContent": "space-between",
        "marginTop": 27,
        "maxWidth": "100%",
      }
    }
  >
    <EmojiButton
      areAnySelected={true}
      emoji="💩"
      isSelected={true}
      key="0"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="😘"
      isSelected={false}
      key="1"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="🤦‍♀"
      isSelected={false}
      key="2"
      onPress={[Function]}
    />
    <EmojiButton
      areAnySelected={true}
      emoji="👿"
      isSelected={false}
      key="3"
      onPress={[Function]}
    />
  </View>
</Fragment>
`;
